lib/yaml_extend.rb

```diff
@@ -49,23 +49,23 @@ def self.ext_load_file(yaml_path, inheritance_key=nil, extend_existing_arrays=tr
     if inheritance_key.nil?
       inheritance_key = @@ext_load_key || DEFAULT_INHERITANCE_KEY
     end
-    total_config ||= {}
+    total_config = config.clone
```

Comment:
Not sure about that, but since we are introducing breaking changes anyways, how do you feel about removing the internal `config` parameter from this function?
That would mean renaming the current function, so that it is used internally, and having the public one drop this parameter from the interface - will only pass `{}` to the first call of the internal function.